python/tank/commands/core_upgrade.py

```diff
@@ -158,7 +159,9 @@ def _run(self, log, core_version):
         log.info("https://support.shotgunsoftware.com/entries/96142347")
         log.info("")
 
-        installer = TankCoreUpdater(code_install_root, log, core_version)
+        config_desc = self.tk.configuration_descriptor if self.tk is not None else None
```

Comment:
Mostly we will get a config descriptor, however I've tested and we don't get one when running on a studio core (which makes sense). I think this is probably the only case, but the code handles not getting passed it anyway.